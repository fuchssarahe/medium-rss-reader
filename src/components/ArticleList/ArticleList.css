.ArticleList {}

.ArticleList--empty {
  text-align: center;
  margin: 2em;
}

.ArticleList--fetching {}
