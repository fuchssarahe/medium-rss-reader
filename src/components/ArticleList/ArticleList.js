import React from 'react';
import PropTypes from 'prop-types';
import Article from '../Article/Article';
import './ArticleList.css';

// Responsible for rendering all articles for a given RSS feed.
// Also responsible for rendering the empty state and intermediate state
// while GET requests are in-flight.
const ArticleList = ({ articles, isFetching = false, addTagFilter }) => {
  if (isFetching) return <FetchingArticlesList />;
  if (!articles || articles.length === 0) return <EmptyArticlesList />;

  return (
    <ul className='ArticleList'>
      {
        articles.map(item => {
          return <Article item={item} addTagFilter={addTagFilter} key={item.link} />
        }
      )}
    </ul>
  );
}

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.object).isRequired,
  isFetching: PropTypes.bool,
  addTagFilter: PropTypes.func.isRequired
}

const FetchingArticlesList = () => {
  return (
    <div className='ArticleList ArticleList--fetching ArticleList--empty'>Fetching articles...</div>
  )
}

const EmptyArticlesList = () => {
  return (
    <div className='ArticleList ArticleList--empty'>There are no articles yet! Better find some!</div>
  )
}

export default ArticleList;
