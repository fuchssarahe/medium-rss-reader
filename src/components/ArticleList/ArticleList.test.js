import React from 'react';
import { shallow } from 'enzyme';
import ArticleList from './ArticleList';

const mockArticles = [{ title: 'Article 1', link: 'a' }, { title: 'Article 2', link: 'b' }]

describe('<ArticleList />', () => {
  it('renders Articles for each provided article object', () => {
    expect(shallow(
      <ArticleList articles={mockArticles} addTagFilter={() => {}} />
    )).toMatchSnapshot();
  })

  describe('when "articles" is undefined', () => {
    test('it renders an empty element', () => {
      expect(shallow(<ArticleList addTagFilter={() => {}} />)).toMatchSnapshot();
    });
  });

  describe('when no articles are present', () => {
    test('it renders an empty element', () => {
      expect(shallow(<ArticleList articles={[]} addTagFilter={() => {}} />)).toMatchSnapshot();
    });
  });

  describe('when articles are currently being fetched', () => {
    test('it renders an empty element', () => {
      expect(shallow(
        <ArticleList
          articles={mockArticles}
          addTagFilter={() => {}}
          isFetching={true}
        />
      )).toMatchSnapshot();
    });
  });
});
