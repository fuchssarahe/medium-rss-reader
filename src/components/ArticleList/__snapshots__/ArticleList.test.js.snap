// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> renders Articles for each provided article object 1`] = `
<ul
  className="ArticleList"
>
  <Article
    addTagFilter={[Function]}
    item={
      Object {
        "link": "a",
        "title": "Article 1",
      }
    }
    key="a"
  />
  <Article
    addTagFilter={[Function]}
    item={
      Object {
        "link": "b",
        "title": "Article 2",
      }
    }
    key="b"
  />
</ul>
`;

exports[`<ArticleList /> when articles are currently being fetched it renders an empty element 1`] = `<FetchingArticlesList />`;

exports[`<ArticleList /> when no articles are present it renders an empty element 1`] = `<EmptyArticlesList />`;
